src/main/native/smithwaterman/PairWiseSW.h

```diff
@@ -428,8 +430,13 @@ int32_t CONCAT(runSWOnePairBT_,SIMD_ENGINE)(int32_t match, int32_t mismatch, int
     int32_t  *E_  = (int32_t *)_mm_malloc((6 * (MAX_SEQ_LEN+ AVX_LENGTH)) * sizeof(int32_t), 64);
     int16_t  *backTrack_ = (int16_t *)_mm_malloc((2 * MAX_SEQ_LEN * MAX_SEQ_LEN + 2 * AVX_LENGTH) * sizeof(int16_t), 64);
     int16_t  *cigarBuf_  = (int16_t *)_mm_malloc(4 * MAX_SEQ_LEN * sizeof(int16_t), 64);
-   
-    if (E_ == NULL  | backTrack_  == NULL | cigarBuf_ == NULL) {
+
+    int len = max(len1, len2);
+    if(len >  MAX_SEQ_LEN) {
```

Comment:
New condition to check input is not too long for `cigarBuf_` or `p->cigar` eventually